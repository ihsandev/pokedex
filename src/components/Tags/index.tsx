import styled from "styled-components";
import { PokemonTypeColor } from "../../utils";

const Tags = styled.span`
  padding: 5px;
  border-radius: 5px;
  color: #fff;
  margin-right: 10px;
  font-size: 0.8em;
  background-color: ${({ color }: any) => PokemonTypeColor(color)};
`;

export default Tags;
