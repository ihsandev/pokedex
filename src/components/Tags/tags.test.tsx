import * as React from "react";
import { shallow } from "enzyme";
import Tags from "./index";

test("render Tags with props", () => {
  const wrapper = shallow(<Tags color="grass">Grass</Tags>);
  expect(wrapper).toMatchSnapshot();
});
