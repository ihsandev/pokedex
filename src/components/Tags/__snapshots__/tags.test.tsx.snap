// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Tags with props 1`] = `
<StyledComponent
  color="grass"
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  padding: 5px;
  border-radius: 5px;
  color: #fff;
  margin-right: 10px;
  font-size: 0.8em;
  background-color: ",
          [Function],
          ";
",
        ],
      },
      "displayName": "styled.span",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "span",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
>
  Grass
</StyledComponent>
`;
